370-auth-ui/oidc.md

```diff
@@ -0,0 +1,351 @@
+
+#### nonce
+
+No difference from the spec.
+
+#### display
+
+It is not supported. The only supported display mode is `page`.
+
+#### prompt
+
+- not specified: Prompt user for authentication or authorization, depends on Idp session state.
+- `none`: No difference from the spec. For requesting new access token when token expire in SPA.
+
+#### max_age
+
+It is not supported.
+
+#### ui_locales
+
+No difference from the spec.
+
+#### id_token_hint
+
+No difference from the spec, for `prompt=none` case.
+
+#### login_hint
+
+It is not supported.
+
+#### acr_values
+
+It is not supported.
+
+#### code_challenge_method
+
+Only `S256` is supported. `plain` is not supported. The reason is the existing implementation of SSO is using `S256` already. Since it defaults to plain, it is required and the value must be `S256`.
+
+#### code_challenge
+
+No difference from the spec.
+
+#### Example of valid Authentication Request
+
+`response_type=code&scope=openid&client_id=CLIENT_ID&redirect_uri=REDIRECT_URI&code_challenge_method=S256&code_challenge=CODE_CHALLENGE&ui_locales=en`
+
+### Token Request
+
+The following sessions list out the request parameters and their difference from the spec.
+
+#### grant_type
+
+Either `authorization_code` or `refresh_token`
+
+#### code
+
+No difference from the spec.
+
+#### redirect_uri
+
+No difference from the spec.
+
+#### client_id
+
+It is always required. The value must be a valid Skygear API Key.
+
+#### code_verifier
+
+No difference from the spec.
+
+#### Example of valid Token Request
+
+`grant_type=authorization_code&code=AUTHORIZATION_CODE&redirect_uri=REDIRECT_URI&client_id=API_KEY&code_verifier=CODE_VERIFIER`
+
+### Token Response
+
+#### id_token
+
+No difference from the spec.
+
+#### token_type
+
+It is always the value `bearer`.
+
+#### access_token
+
+No difference from the spec.
+
+#### expires_in
+
+No difference from the spec.
+
+#### refresh_token
+
+No difference from the spec. Absent in [SPA](#case-2-single-page-web-app-spa) flow.
+
+#### scope
+
+It is always absent.
+
+#### Example of Token Response
+
+```json
+{
+  "token_type": "bearer",
+  "access_token": "access.token",
+  "refresh_token": "refresh.token"
+}
+```
+
+## Configurations
+
+### Sample configuration
+
+```yaml
+app_config:
+  clients:
+  - client_id: XXX
+    client_secret: XXX
+    session_lifetime: 86400
+    access_token_lifetime: 1800
+    session_idle_timeout_enabled: true
+    session_idle_timeout: 300
+    client_name: Mobile App
+    application_type: native
+    redirect_uris:
+    - "https://example.com"
+    logo_uri: "https://example.com/logo.png"
+    grant_types:
+    - "authorization_code"
+    - "refresh_token"
+    response_types:
+    - "code"
+```
+
+### Parameters
+
+#### OIDC client Metadata
+
+Some parameters are defined in OIDC, see [ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata).
+
+They are
+
+- `client_name`
+- `application_type`
```

Comment:
This is included in OIDC spec but [removed in OAuth](https://tools.ietf.org/html/draft-ietf-oauth-dyn-reg-30). Actually we do not need it?

For the portal use case, we offer a "profile" for the developer to choose from. The profiles include

Server side relying party

```yaml
grant_types: ["authorization_code", "refresh_token"]
response_types: ["code"]
```

Native applications (including mobile applications and desktop applications)

```yaml
grant_types: ["authorization_code", "refresh_token"]
response_types: ["code"]
```

Single page application

```yaml
grant_types: ["authorization_code"]
response_types: ["code"]
```

Server side rendering application

```yaml
grant_types: []
response_types: ["none"]
```

Reply:
Yes, OP actually don't read it . I removed it